### PR TITLE
Add slice import 

### PR DESCRIPTION
**Pull request by ForsakenHarmony**
New version of #192 potentially a bit better

The error name is maybe not ideal

### CODE REVIEW

File: src/import.rs [hunk 1]
```diff
@@ -203,5 +201,32 @@ fn import_impl(path: &Path) -> Result<Import> {
 pub fn import<P>(path: P) -> Result<Import>
     where P: AsRef<Path>
 {
-    import_impl(path.as_ref())
+    let path = path.as_ref();
+    let base = path.parent().unwrap_or(Path::new("./"));
+    let file = fs::File::open(path).map_err(Error::Io)?;
+    let reader = io::BufReader::new(file);
+    import_impl(Gltf::from_reader(reader)?, Some(base))
```
Comment (alteous) [inline src/import.rs]: We need to keep this similiar to before or else the compiler will generate lots of code, once for every `P` type. I suggest we add another intermediate function:

```rust
// New function (name could be improved!)
fn import_path(path: &Path) -> Result<Import> {
    ...
    import_impl(Gltf::from_reader(reader)?, Some(base))
}

// As you have
fn import_impl(Gltf { document, blob }: Gltf, base: Option<&Path>) -> Result<Import> {
    ...
}

// As before
pub fn import<P>(path: P) -> Result<Import>
    where P: AsRef<Path>,
{
    import_path(path.as_ref())
}
```

We could also make the slice parameter more generic:
```rust
fn import_slice_impl(slice: &[u8]) -> Result<Import> {
    import_impl(Gltf::from_slice(slice)?, None)
}

pub fn import_slice<S>(slice: S) -> Result<Import>
    where S: AsRef<[u8]>,
{
    import_slice_impl(slice.as_ref())
}
```

File: src/lib.rs [hunk 2]
```diff
@@ -107,7 +107,7 @@ pub mod image;
 
 /// The reference importer.
 #[cfg(feature = "import_data_reference")]
-mod import;
+pub mod import;
```
Comment (alteous) [inline src/lib.rs]: Could we avoid making this module public?

File: src/lib.rs [hunk 3]
```diff
@@ -207,6 +210,10 @@ pub enum Error {
     #[cfg(feature = "import")]
     MissingBlob,
 
+    /// An external file is referenced in a slice only import without path
+    #[cfg(feature = "import")]
+    ExternalReferenceInSliceImport,
```
Comment (alteous) [inline src/lib.rs]: This is a reasonable approach.

File: src/lib.rs (hunk 2 above)
Comment (ForsakenHarmony) [reply src/lib.rs]: Yeah, not sure why I had that in there

File: src/import.rs (hunk 1 above)
Comment (ForsakenHarmony) [reply src/import.rs]: Done I think 👌

**Comment by ForsakenHarmony** [comment]
🎉

**Pull request merged**